test/integration/switch_int_test.go

```diff
@@ -0,0 +1,111 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/ActiveState/cli/internal/constants"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/ActiveState/cli/pkg/projectfile"
+	"github.com/stretchr/testify/suite"
+)
+
+type SwitchIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *SwitchIntegrationTestSuite) TestSwitch_Branch() {
+	suite.OnlyRunForTags(tagsuite.Switch)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	err := ts.ClearCache()
+	suite.Require().NoError(err)
+
+	suite.PrepareActiveStateYAML(ts, "ActiveState-CLI", "Branches")
+	pjfilepath := filepath.Join(ts.Dirs.Work, constants.ConfigFileName)
+
+	cp := ts.SpawnWithOpts(e2e.WithArgs("pull"), e2e.AppendEnv(constants.DisableRuntime+"=false"))
```

Comment:
I don't think you need the runtime to test this?

Reply:
Also can we get rid of the pull and just bake the commitID into the test? For tests it's better that way, costs less time and gives fewer things that can break. Right now you could implicitly break the test by adding a new commit to the project.

Reply:
We're using the runtime in order to ensure that it is refreshed. We don't test the runtime, just the output associated with refreshing it. I don't believe that mechanism will give us the desired output without both enabling the runtime and the initial pull to install the runtime files.

Reply:
I could definitely be wrong here, but I think you can safely disable it.

While I agree it would be good to test that the runtime did in fact change it is also quite costly in terms of how long it takes to test. Seeing as we already have tests for this mechanic I would advocate taking it out of this one. 

Perhaps we can set up something for testing purposes that allows us to assert this without the cost of setting up a full runtime. I've filed a story to investigate: https://activestatef.atlassian.net/browse/DX-1173